code/modules/power/lighting.dm

```diff
@@ -117,6 +118,9 @@
 		if (src in stationLights)
 			stationLights -= src
 
+		if (inserted_lamp)
+			qdel(inserted_lamp)
```

Comment:
```suggestion
		if (inserted_lamp)
			qdel(inserted_lamp)
			inserted_lamp = null
```
When you are deleting stuff you should clean up all references to it so it's disposed properly. And `disposing` should just clear up all references to things no matter what.